soql-standalone-parser/src/main/scala/com/socrata/soql/BinaryTree.scala

```diff
@@ -0,0 +1,90 @@
+package com.socrata.soql
+
+trait BinaryTree[+T] {
+
+  def seq: Seq[T] = {
+    Seq(asLeaf)
+  }
+
+  /**
+   * output schema
+   */
+  def previous: T = {
```

Comment:
This naming seems weird.

Reply:
any suggestion, outputSchemaLeaf?

Reply:
Sure, that sounds good!  Any hint in the name that the purpose of the method is to say "what is the ultimate schema that this subtree will produce".